app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt

```diff
@@ -522,13 +558,110 @@ class FeedFragment : BaseStateFragment<FeedState>() {
         )
     }
 
+    /**
+     * Highlights all items that are after the specified time
+     */
+    private fun highlightNewItemsAfter(updateTime: OffsetDateTime) {
+        var highlightCount = 0
+
+        var doCheck = true
+
+        for (i in 0 until groupAdapter.itemCount) {
+            val item = groupAdapter.getItem(i) as StreamItem
+
+            var typeface = Typeface.DEFAULT
+            var resid = R.attr.selectableItemBackground
+            if (doCheck) {
+                if (item.streamWithState.stream.uploadDate?.isAfter(updateTime) != false) {
+                    typeface = Typeface.DEFAULT_BOLD
+                    resid = R.attr.dashed_border
+                    highlightCount++
+                } else {
+                    // Increases execution time due to the order of the items (newest always on top)
+                    // Once a item is is before the updateTime we can skip all following items
+                    doCheck = false
+                }
+            }
+
+            // The highlighter has to be always set
+            // When it's only set on items that are highlighted it will highlight all items
+            // due to the fact that itemRoot is getting recycled
+            item.execBindEnd = Consumer { viewBinding ->
+                val context = viewBinding.itemRoot.context
+                viewBinding.itemRoot.background =
+                    androidx.core.content.ContextCompat.getDrawable(
+                        context,
+                        android.util.TypedValue().apply {
+                            context.theme.resolveAttribute(
+                                resid,
+                                this,
+                                true
+                            )
+                        }.resourceId
+                    )
+                viewBinding.itemVideoTitleView.typeface = typeface
+            }
+        }
+
+        // Force updates all items so that the highlighting is correct
+        // If this isn't done visible items that are already highlighted will stay in a highlighted
+        // state until the user scrolls them out of the visible area which causes a update/bind-call
+        groupAdapter.notifyItemRangeChanged(
+            0,
+            groupAdapter.itemCount.coerceAtMost(highlightCount.coerceAtLeast(lastNewItemsCount))
```

Comment:
Wouldn't this be more readable as `min(groupAdapter.itemCount, max(highlightCount, lastNewItemsCount))`? I had to look up the meaning of `coerceAt*`. Just an opinion from someone who doesn't code much in Kotlin, so feel free to contradict me ;-)

Reply:
First I had it like this but than Android studio said:
![grafik](https://user-images.githubusercontent.com/40789489/132907266-3b9bd9a1-9fc9-4da0-9cbb-ab50195f518e.png)

However I found out that in kotlin you can actually use ``minOf`` and ``maxOf`` which is far better that ``coerceAt*`` so I will use that instead 😄

Reply:
Ok, perfect, thank you